backend/sources/facebook/connector/src/main/java/co/airy/core/sources/facebook/api/Mapper.java

```diff
@@ -3,28 +3,43 @@
 import co.airy.avro.communication.Message;
 import co.airy.core.sources.facebook.api.model.SendMessagePayload;
 import co.airy.core.sources.facebook.dto.SendMessageRequest;
+import co.airy.mapping.ContentMapper;
+import co.airy.mapping.model.Content;
+import co.airy.mapping.model.SourceTemplate;
+import co.airy.mapping.model.Text;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+
 
 @Service
 public class Mapper {
+    private final ContentMapper mapper;
 
-    private final ObjectMapper objectMapper;
-
-    Mapper(ObjectMapper objectMapper) {
-        this.objectMapper = objectMapper;
+    Mapper(ContentMapper mapper) {
+        this.mapper = mapper;
     }
 
     public SendMessagePayload fromSendMessageRequest(SendMessageRequest sendMessageRequest) throws Exception {
         final Message message = sendMessageRequest.getMessage();
 
         final SendMessagePayload.MessagePayload messagePayload = new SendMessagePayload.MessagePayload();
 
-        final JsonNode messageRequest = objectMapper.readTree(message.getContent());
-
-        messagePayload.setText(messageRequest.get("text").textValue());
+        final Content content = mapper.render(message)
+                .stream()
+                .findFirst()
+                .orElseThrow(() -> new Exception(""));
+
+        if (content instanceof Text) {
+            messagePayload.setText(((Text) content).getText());
+        } else if (content instanceof SourceTemplate) {
+            messagePayload.setAttachment(SendMessagePayload.AttachmentPayload.builder()
+                    .type("template")
+                    .payload(((SourceTemplate) content).getPayload())
+                    .build());
+        }
```

Comment:
We can use a `Map<String,Function>` here
It's a pattern we used on the source mappers and I think they could be useful here

Reply:
How would that work in this case; i.e. what would be the key?

Reply:
Hm, good point

Do we have a `type` that we can use? If not maybe a `String` key that is the Class name
So a map could be

`"Text" -> renderingTextFunction`

`renderingTextFunction` is a Function that takes a MessagePayload and returns a MessagePayload

```
    private static final Function<MessagePayload, MessagePayload> renderingTextFunction = (messagePayload) -> <Return new modified message payload>;
```

What do you think?

Reply:
Unfortunately, we have neither, because we have to check against the instance. The string type key only exists after serialization. In any case for an _n=2_ I think the code would end up looking more complex than it is now.